### ISSUE TITLE
Debug events issue on the first statement of the begin block in jruby 9.4

### ISSUE THREAD

**Issue by hurricup** [jruby/jruby#7495]
When trying to debug a ruby script with [ruby-debug](https://github.com/ruby-debug/ruby-debug) it now stops twice on first statement on the begin block. 

```
begin 
  puts 42 #bp here, will stop second time after stepping over
  puts 42
rescue TypeError => e
  puts "Name error"
rescue
  puts "rescued"
ensure
  puts("ensured")
end
```

I'm not really familiar on how this work in jruby, could not even introspect iseq / events. So can't be much help here yet. But probably it will hint you about something. 

Works fine in < 9.4

Tested from RubyMine

**Comment by hurricup** [comment]
This was fixed in 9.4.7. Couldn't find the change, but this double step is no longer happening. Also, debugger is now stopping in begin itself (consistent with pre-9.4 behavior)

**Comment by headius** [comment]
Thanks for the follow-up! This was probably fixed along with other fixes in execution tracing from @enebo.
